lib/rules/indent.js

```diff
@@ -1217,8 +1217,10 @@ module.exports = {
                 const firstNonObjectToken = sourceCode.getFirstTokenBetween(node.object, node.property, astUtils.isNotClosingParenToken);
                 const secondNonObjectToken = sourceCode.getTokenAfter(firstNonObjectToken);
 
-                const tokenBeforeObject = sourceCode.getTokenBefore(node.object, token => astUtils.isNotOpeningParenToken(token) || parameterParens.has(token));
-                const firstObjectToken = tokenBeforeObject ? sourceCode.getTokenAfter(tokenBeforeObject) : sourceCode.ast.tokens[0];
+                const objectParenCount = sourceCode.getTokensBetween(node.object, node.property).findIndex(astUtils.isNotClosingParenToken);
```

Comment:
Updated to use `filter` and `length`. I kept using `getTokensBetween` rather than `getFirstTokensBetween`, because I think there's no difference between them when no count is provided.